lib/rspec/matchers/built_in/contain_exactly.rb

```diff
@@ -33,6 +33,13 @@ def description
 
       private
 
+        def reset
+          @pairings_maximizer = nil
+          @best_solution = nil
+          @extra_items = nil
+          @missing_items = nil
+        end
```

Comment:
```suggestion
        def matches?(actual)
          @pairings_maximizer = nil
          @best_solution = nil
          @extra_items = nil
          @missing_items = nil
          super(actual)
        end
```

Reply:
@JonRowe Yep, this is even nicer than my approach: it keeps the change local to `ContainExactly`.  Just incorporated.  Mind kicking off a build?

Reply:
One more build and should be good to go!